model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanSingleUseObjectProvider.java

```diff
@@ -61,9 +67,14 @@ public void put(String key, long lifespanSeconds, Map<String, String> notes) {
             if (logger.isDebugEnabled()) {
                 logger.debugf(re, "Failed when adding code %s", key);
             }
-
             throw re;
         }
+        if (persistRevokedTokens && key.endsWith(REVOKED_KEY)) {
+            if (!notes.isEmpty()) {
+                throw new ModelException("notes are not supported for revoked tokens");
```

Comment:
Should we also throw an exception in all other methods in this class? For example, when `remove` method is called with the `REVOKED_KEY` suffix the operation is not propagated to the database and entries can be reintroduced after a restart.

Reply:
Done. I think this API is due for a redesign. I combined the check with the `persistRevokedTokens` for now to make it possible for people to opt-out on this.